### DISCUSSION TITLE
some zsh completions don't work on Apple Silicon Macs while they do on Intel Macs

### DISCUSSION THREAD

**Discussion by jefftemplon** [Homebrew/discussions#5185] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: d7d4c8266210c024b93a450a7d357cec0b46a1bb
Last commit: 4 days ago
Core tap JSON: 08 Mar 09:21 UTC
Core cask tap JSON: 08 Mar 09:21 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: mg
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: /usr/bin occurs before /opt/homebrew/bin in your PATH.
This means that system-provided programs will be used instead of those
provided by Homebrew. Consider setting your PATH so that
/opt/homebrew/bin occurs before /usr/bin. Here is a one-liner:
  echo 'export PATH="/opt/homebrew/bin:$PATH"' >> ~/.zshrc

The following tools exist at both paths:
  openssl
  pip3
  python3
```

### Description of issue

I noticed firstly that brew command completion was not working on my Apple Silicon machine, while it was on an older intel Mac machine.  Later I saw that the same was true for cargo (installed via homebrew).  I could see in the .zcompdump files that for the intel Mac, _brew and _cargo were mentioned while not for the .zcompdump from the Apple Silicon Mac.  I deleted and reinstalled brew, also deleted and reinstalled oh-my-zsh.  I also made the permissions changes to the zsh site-functions directories in /opt/homebrew.  Finally on a hunch, I made a symlink:

`ln -s /opt/homebrew/share/zsh/site-functions /usr/local/share/zsh/`

thereafter completions are working.  I have verified that it's not a problem with my .zshenv / .zshrc files - there is no reference there to /usr/local - there IS a reference to /usr/local in some of the configuration files in .oh-my-zsh, I think there lies the problem.

**Comment by branchv** [comment]
See https://docs.brew.sh/Shell-Completion#configuring-completions-in-zsh, you'll need to update `FPATH`

**Comment by jefftemplon** [comment]
Thank you very much.  I am guessing that on Homebrew for Intel Macs, oh-my-zsh has some hardcoded path to /usr/local/ that causes it to "just work" and not on an intel Mac.
